### ISSUE TITLE
Airwin2Rack

### ISSUE THREAD

**Issue by baconpaul** [VCVRack/community#751]
Hi!

With Chris / @airwindows at [airwindows](https://www.airwindows.com) I've been working on bringing the entire open source air windows effect collection to VCV rack in a single module. We finally got this working and read ready for the library. Last week @cschol confirmed it built in the docker toolchain, and our CI builds it in qnos toolchain image, so should all be good

Details:
Repository: https://github.com/baconpaul/airwin2rack
~~Version: 2.0.0
Hash: fee1cc39966158cdaa84e2969e0ffdc58dce6c65~~ (See below for update)

After working with Chris he asked that we use the brand "Airwindows" for these plugins, which is his brand. I know you have great ethics rules about brand usage, but you can see Chris being active https://community.vcvrack.com/t/airwindows-a-fresh-approach-looking-for-some-testers-and-feedback/19182/11?u=baconpaul in our discussion of the item. But for avoidance of any doubt I asked him explicitly on our discord where we've been working and he shared the following:

<img width="685" alt="Screen Shot 2023-01-23 at 9 13 52 PM" src="https://user-images.githubusercontent.com/13387561/214199062-ce7d37b6-b19b-43b5-afcd-6fc5d9521f0f.png">

Please let us know if you need anything to help get this module into the library! And thanks as always for all your efforts.

**Comment by airwindows** [comment]
Confirmed: this is me, delegating some of the mechanics for making and supporting an ongoing Rack port, through use of my MIT-licensed codebase, and I'd like the project to use the brand 'Airwindows'. The 'entire open source Airwindows effect collection' is defined as also including new stuff as it comes out, and that looks practical to do, so this is not a one-time snapshot, my involvement is in continuing to develop effects and feeding them into the Airwin2Rack port machine regularly.

Looking forward to this! :)

**Comment by baconpaul** [comment]
yeah just from a cadence perspective this one will probably have 6+ updates a year in the library. It's all automated as Chris adds effects but i won't push every new effect as a library version since they do require some manual work from you. But those updates shouldn't change anything other than adding new effects to the menus and options. If there's a cadence that works better for you please let me know.

Thanks!

**Comment by baconpaul** [comment]
Oh also this builds both x86 and ARM no problem on mac. (In fact I did almost all my work with it in ARM).

**Comment by airwindows** [comment]
As long as I can direct people to the dailies on all platforms, so users CAN have the very newest plugin in their Rack each week if they want it, it's fine with me if the library doesn't update every single time. It can be a lot :) never more than weekly, as a rule…

**Comment by baconpaul** [comment]
yeah i still have one button to press when you update so we will lag a smidge but its an easy button to press so we can work that out. lets chat on discord about that not here.

**Comment by baconpaul** [comment]
Hi Folks

We found a couple of bugs in the last week so would prefer to submit version 2.0.1

version: 2.0.1
hash: af3975bca3484381059409008024525d83dba237
repo: https://github.com/baconpaul/airwin2rack

I ran the rack plugin library validator over it; for some reason the authorURL of https://www.airwindows.com/ doesn't validate on my mac. It is a legit URL though. Don't know if this is a problem.

Thanks as always.

**Comment by airwindows** [comment]
Given that airwindows.com is my website, if there's a problem with it validating with something I'm sure I can fix it :)

**Comment by Petervos2018** [comment]
SOLVED. site is back up 👍 

As of 27012023 the site is not reachable with firefox or edge, did you try  it yourself ?

![afbeelding](https://user-images.githubusercontent.com/46076474/215125278-8142bf6d-1349-4eb7-8361-c549d31f9e52.png)

**Comment by baconpaul** [comment]
Even with the site up I get this

```
paul ~/dev/music/rack/rack-library-tools (master) % python3 ./rack-manifest-validator.py -p airwin2rack ..
[airwin2rack] Validating plugin.json...
Exception validating URL: https://www.airwindows.com/ (HTTP Error 406: Not Acceptable)
FAILED
[airwin2rack] Issues found in `plugin.json`:

Invalid URL: https://www.airwindows.com/
```

even though I'm looking at the site now. I think this is something in the interaction between the validator and the website and was meant as a comment for school though really.

**Comment by airwindows** [comment]
What does the validator even want? Do I need to tell my site security to allow something it's not allowing? Whitelist something?

**Comment by baconpaul** [comment]
I don't know! I'm sure when the rack team turns to this they will tell us though.

My original guess was it was doing some user agent filtering or has a robots.txt but we need cschol to tell us for sure. Or just proceed with the exception also, which is fine.

**Comment by cschol** [comment]
The validator is just looking for a valid website by trying to access the URL. I have seen issues with the Python library using is because it doesn't understand redirects, etc. Don't worry about this failure here. I need to look at a more robust way to check if the website is valid (which it obviously is).

**Comment by cschol** [comment]
Your plugin has been integrated and will be available for all supported platforms with the next Plugin Manager update. Please let us know in **this** issue when a new version is available. We will reopen it. Thank you for providing your plugin to the Rack community.

**Comment by baconpaul** [comment]
Thanks @cschol as always for your work on the library!

**Comment by baconpaul** [comment]
Hi!

I'm not going to update this plugin in the library for each and every release of a chris effect - we have nightlies for that - but I would like to try the roundtrip with a new version once, and chris just added a new reverb.

So if we could update

Version: 2.1.0
Hash: 045b7dd9f6837097b5c2001cd5e7cf44edbeac50

that would be great. I'll probably then hop to an every 4-6 week cadence on library pushes.

Thanks

**Comment by baconpaul** [comment]
Hi! 4 new fx this month.

Version 2.2.0
Hash 87a2bd679d5bedfb00a151862ef1e5161effddb9

Thanks!

**Comment by baconpaul** [comment]
Hi!

Version: 2.3.0
Hash: 893bbe235f16f6d437278f41cd937a7254162c4f

Thanks!

**Comment by baconpaul** [comment]
Happy June!

Version: 2.4.0 
Hash: 899424daa4035d8e88496c5d485dedeadbc1a901

Thanks as always!

**Comment by AndrewBelt** [comment]
This plugin builds with Rack SDK 2.4.0 for Mac ARM64, but it doesn't load due to not finding the `init()` symbol. I'm not sure why that would be, but it could be an easy fix.

**Comment by baconpaul** [comment]
Ok I’ll take a look this week! Thanks for the heads up

**Comment by baconpaul** [comment]
OK I confirmed that with the rack 2.3 SDK I get an init and a loadable plugin arm64 and with the rack 2.4 SDK I do not.

Let me dig a bit deeper. I'll also tag @qno in case he has seen similar.

Thanks Andrew! Will revert once I have more.

**Comment by baconpaul** [comment]
Ahh yeah I found it

We were building `plugin-arm64.dylib` in our cmake wrapper and that changed back to `plugin.dylib` in the 2.4.0 sdk. Will have a fix together for this (and also BaconMusic and Surge which will need it).

Might not get all three today. we have no power and are on generator. but will get to em this week.

Thanks!

**Comment by baconpaul** [comment]
OK

version: 2.5.0
hash: bc39556738e0bf27af1390d49aa77f2e0b886041

as well as adding a variety of effects, it also remediates this arm problem

I'll prep versions of surge and bacon music that do the same but if you could confirm that has builds and loads for you arm that would be great. Thank you so much.

**Comment by baconpaul** [comment]
Ok versions of all 3 are now in library. Thanks!

If I may suggest this line in the changelog

> Move plugins folder to `<Rack user folder>/plugins-<OS>-<CPU>.`

maybe expand it to be

> Move plugins folder to `<Rack user folder>/plugins-<OS>-<CPU>;` Revert 2.2.2 change where arm libraries are named `plugin-arm64.dylib` returning to `plugin.dylib` all architectures

**Comment by baconpaul** [comment]
Hi folks!

Small tweak to fix one bug

Version: 2.5.1
Hash: 57a9cc68344107c7ae50f6644c9bdfab87de4c97

Thanks as always!

**Comment by baconpaul** [comment]
Morning folks

v2.6.0
72d08d50fe660b6f921e36e81ba51efa4f36958f

ready for the library.

Thanks so much!

**Comment by baconpaul** [comment]
Hi Folks!

2.7.0 ready to go

Hash: ed85c825d7c493f3651c207df273f6024e91dd60
Tag: v2.7.0

Thanks so much!

**Comment by baconpaul** [comment]
Hello!

Version 2.8.0
Hash: d1eb0db0042cadfa6f32fc4e5146e4e0e719b939

Thanks!

**Comment by baconpaul** [comment]
v2.9.0 is ready

hash a93d0a33f71bf34160875f63cf0129c187be0899

Thanks!

**Comment by baconpaul** [comment]
Hi!

~~2.10.0 is ready~~

~~24a3dea142924b0199bd0a87e52d53069c3bc9e3~~

EDIT: Please hold off. 2.10.0 has a bug we just found. I'll have a 2.11.0 soon

We've made some changes in the repo a bit bigger than normal. I tested everything in the docker image and got a clean build, and was sure to isolate those changes from the rack-facing side of the project, but if you run into any problems let me know of course.

Thanks!

**Comment by baconpaul** [comment]
Hi

sorry about that false start

2.11.0
46861020727eaddc66a4aa021865d20fc36ef3c1

Thanks so much!

**Comment by baconpaul** [comment]
Hi folks!

2.12.0
e4dcdf547139ef5e94ed7b3a72ce5bd81148aba9

Thanks!

**Comment by baconpaul** [comment]
Hi folks!

2.13.0
d288900a0049725e9018f30d78c7d6c8dc230eab

Thanks!

**Comment by AndrewBelt** [comment]
Re-built using our fixed toolchain https://github.com/VCVRack/rack-plugin-toolchain. Let us know if users are still experiencing issues loading this plugin on Windows.

**Comment by baconpaul** [comment]
Thanks Andrew!

**Comment by baconpaul** [comment]
Getting crash reports this morning both on Airwindows and TinyTricks from windows user. Both mac is fine and the nightly built with the gcc 13 toolchain is fine.

So I think we aren't out of the woods.

**Comment by AndrewBelt** [comment]
5 Windows users have tested this library build with success, but let me know if it doesn't load and I'll revert again.
